components/raftstore/src/store/entry_storage.rs

```diff
@@ -479,13 +479,7 @@ fn validate_states<ER: RaftEngine>(
         info!("updating commit index"; "region_id" => region_id, "old" => commit_index, "new" => recorded_commit_index);
         commit_index = recorded_commit_index;
     }
-    // Invariant: applied index <= max(commit index, recorded commit index)
-    if apply_state.get_applied_index() > commit_index {
-        return Err(box_err!(
-            "applied index > max(commit index, recorded commit index), {}",
-            state_str()
-        ));
-    }
+
```

Comment:
i think its good to print the info message during boot. It would be helpful in debugging.

Reply:
I would add assert that term of applied indx and commit indx has to be same.

Reply:
> i think its good to print the info message during boot. It would be helpful in debugging. 
Yes, good suggestion. Replaced the error with an info log.

> I would add assert that term of applied indx and commit indx has to be same.
Yes, there is already a check for `raft_state.get_hard_state().get_term() < apply_state.get_commit_term()`. While `apply_state.get_commit_term()` is not the exact the term of the latest applied log because we don't persist this term.